docs/user_authentication/third_party_token_management.md

```diff
@@ -0,0 +1,40 @@
+
+```sequence {theme: 'simple'}
+title Third Party IDP Token Management
+
+participant Pbench-Keycloak-server
+participant Pbench-Server
+participant Pbench-Dashboard
+participant Identity-Provider
+
+autonumber 1
+Pbench-Dashboard->Pbench-Server: GET login page
+Pbench-Server->Pbench-Dashboard: 200 login page
+Pbench-Dashboard->Identity-Provider: POST Authorization request
+note over Identity-Provider: Authenticate User and request \nuser consent for Pbench
+Identity-Provider->Pbench-Dashboard: 200 Authorization code response
+Pbench-Dashboard->Identity-Provider: POST authorization code (Token request)
+Identity-Provider->Pbench-Dashboard: 200 Access/Refresh Token payload response 
+note over Identity-Provider,Pbench-Dashboard: {"access_token":"<access_token>",\n"expires_in": 300,\n"refresh_expires_in": 1800,\n"refresh_token": "<refresh_token>",\n"token_type": "Bearer",\n"id_token": "<id_token>",...}
+// Here the dashboard sends the newly acquired access token to the Pbench Server to start the token exchange process.
+Pbench-Dashboard->Pbench-Server: POST /api/v1/pbench-token request (Bearer: Access Token 1)
+Pbench-Server->Identity-Provider: GET public Key
+Identity-Provider->Pbench-Server: 200 Public key
+Pbench-Server->Pbench-Server: Validate the access token
```

Comment:
Yes ideally we should validate the tokens locally and we get all the user information from decoding the token so we don't need to go to the `userinfo` endpoint in most cases. That's why I added the step as an `optional` just in case we need to, that's an option.